File: 242/js/242.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    // check
    let sl = s.length;
    let tl = t.length;
    if (sl !== tl){
        return false;
    }

    // make map
    let mapT = new Map();
    for (const chT of t){
        const count = mapT.get(chT) || 0;
        mapT.set(chT, 1+count);
    }

    // check anagram
    for (const chS of s){
        const count = mapT.get(chS) || 0;
        if (count <= 0){
            return false;
        }else{
            mapT.set(chS, count-1);
        }
    }

    return true;
};

File: 242/go/242.go
func isAnagram(s string, t string) bool {
    // check 
    if len(s) != len(t){
        return false
    }

    // make a map for t, if find t dump it
    mapT := make(map[rune]int, len(t))
    for _, chT := range t {
        mapT[chT] += 1
    }

    // check anagram
    for _, chS := range s{
        mapT[chS] -= 1
        if mapT[chS] < 0 {
            return false
        }
    }

    return true
}

// rune --> range str
// byte --> str[i]